File: src/Tiger.java
public class Tiger extends Feline {


   Tiger(){
      super(false, "Carnivore", "Forests", true, "Teddy", 4);
   }

   /**
    *  Constructor with only Animal parameters
    * @param awake:boolean - Is the animal awake?
    * @param diet:String - What does the animal eat?
    * @param habitat:String - Where does the animal live?
    * @param hungry:boolean - Is the animal hungry?
    * @param name:String - The name of the animal
    * @param numLegs:int - How many legs does the animal have?
    */
   Tiger(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
      super(awake, diet, habitat, hungry, name, numLegs);
   }

   /**
    * Override for makeNoise function
    * 1/3 chance to roar
    * 1/3 chance to chuff ( it's like the tiger equivalent of purring
    * 1/3 chance to growl
    */
   @Override
   public void makeNoise() {
      double randomNumber = Math.random();
      if(randomNumber < 0.33) {
         System.out.println(name + " the " + this.getClass().getName() + ": Roared");
      } else if (randomNumber < 0.66){
         System.out.println(name + " the " + this.getClass().getName() + ": Chuff Chuff");
      } else {
         System.out.println(name + " the " + this.getClass().getName() + ": Growled");
      }
   }
}

File: Project2/Tiger.py
import random
from Feline import Feline

class Tiger(Feline):

  '''
    Generic initializer for Tiger
    @param name:String, default:'Tony'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
  '''
  def __init__(self, name='Tony', awake=False, hungry=True, num_legs=4):
    super(Tiger, self).__init__(name, awake, hungry, num_legs)

  '''
    A tiger will make a noise with the following probability
      - 25% chance to purr
      - 25% chance to grumble 
      - 50% chance to make a CHUFF sound
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if(random_number < 0.25):
      print('{} the {} purred.'.format(self.name, animal_type))
    elif(random_number < 0.5):
      print('{} the {} grumbles.'.format(self.name, animal_type))
    else:
      print('{} the {} makes a CHUFF sound.'.format(self.name, animal_type))
